docs/backend/guides/service-objects/structure.md

```diff
@@ -0,0 +1,145 @@
+
+Our service objects are composed of an interface that is defined in the `services` package as well as a subpackage 
+that contains the implementation logic, validation logic, and tests. This pattern is useful to avoid import cycles. 
+It also keeps a given service's functionality grouped together in a subpackage with `services` without having to put 
+all our code in one huge pile. Finally, interfaces are the only objects in Go that can be effectively mocked, so 
+this is advantageous for our testing.
+
+:::caution Stuttering
+You might notice that there's a lot of redundancy in our naming scheme. This is commonly referred to as 
+**"stuttering"** and is considered an anti-pattern in most languages and frameworks.
+
+We may at some point try to move away from this convention, but it is preferable to be consistent with our 
+less-than-ideal naming scheme for now.
+:::
+
+## Data/Utility Service Objects High Level Structure
+
+Data and utility service objects follow the same structure pattern, with just naming being different, so they'll be 
+covered together at first.
+
+In the `mymove` `./pkg/services` directory you should see several subdirectories (`Go` subpackages) and singular 
+`Go` files at the end. The names of these solo `Go` files should match that of a directory above.
+
+```text {7,11}
+mymove/
+├── pkg/
+│   ├── services/
+│   │   ├── ...
+│   │   ├── mto_agent/
+│   │   ├── mto_service_item/
+│   │   ├── mto_shipment/       <- a subpackage
+│   │   ├── ...
+│   │   ├── mto_agent.go
+│   │   ├── mto_service_item.go
+│   │   ├── mto_shipment.go     <- the interface in the services package
+│   │   ├── ...
+```
+
+### Data Service Objects High Level Structure
+
+The names of the file + subpackage should match the database table name, but in singular form. E.g. `mto_shipment` 
+service object for the `mto_shipments` table.
+
+Each of these subpackages handles the operations for one particular data model. This helps us keep track of our
+database interactions and allows for different parts of the codebase to interact with models while re-using the same 
+validation and functionality.
+
+### Utility Service Objects High Level Structure
+
+The names of the file + subpackage should match the function the service object serves, e.g. `fetch`.
+
+Each of these subpackages should focus on the functionality they're meant to serve. It helps us have logic we can 
+re-use across service objects.
+
+## Orchestrator Service Objects High Level Structure
+
+Orchestrator service objects are similar overall to the other types of service objects, but the top level structure 
+and naming are slightly different. 
+
+We have an `orchestrators` subpackage inside `services` that holds the subpackages for each orchestrator. This helps 
+us separate out the `orchestrators` from the other more "regular" service objects (i.e. pre-existing service object 
+types that existed before `orchestrators` were added). 
```

Comment:
**Question**: I'm reading this as the difference between an orchestrator and a regular service object is whether it's a new service object or not? Asking because I thought there were other differences between the two?

Reply:
Orchestrators are still their own service objects, but the main differences are their placement (inside `orchestrators` subpackage), naming (orchestrators have a bit less stuttering, as our docs call it, and the interface file ends with `_orchestrator.go`), and what type of work they do (light audits and mostly chaining other service objects appropriately). 

Thanks for bringing it up though, I think I can maybe highlight some of these differences more in a few parts.